src/sobesity/domain/services/generate_questions.py

```diff
@@ -0,0 +1,84 @@
+import logging
+import random
+import json
+
+from sobesity.domain.entities import (
+    SkillEntity,
+    SkillFilterEnitity,
+    SkillId,
+    QuestionEntity,
+    QuestionFilterEnitity,
+    QuestionId,
+    AnswerEntity,
+    AnswerFilterEnitity,
+    AnswerId,
+    TypeEntity,
+    TypeFilterEnitity,
+    TypeId,
+)
+from sobesity.domain.serializers import NotFoundSerializer
+from sobesity.domain.utils.response import bad_request_maker
+
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+
+class QuestionGeneratorService:
+    def __init__(self, question_service, skill_service, answer_service, type_service):
+        self.question_service = question_service
+        self.skill_service = skill_service
+        self.answer_service = answer_service
+        self.type_service = type_service
+
+    def generate_questions(self, theme, quantity):
+        skill = self.skill_service.get_list(SkillFilterEnitity(names=[theme]))
+
+        if skill:
+            logger.info(f"Fined skill: '{skill}'")
+            all_questions = self.question_service.get_list(
+                QuestionFilterEnitity(skill_ids=[skill[0].skill_id])
+            )
+            logger.info("Take questions")
+            if not all_questions:
+                return bad_request_maker(
```

Comment:
Raise the error here and process the error in view. It's view's processor function